backend/app/controllers/chat_rooms_controller.rb

```diff
@@ -12,20 +12,22 @@ def create
     render json: { errors: e.record.errors.full_messages }, status: 400
   end
 
+  def add_user
+    @chat_room = ChatRoom.find(params[:id])
+    user = User.find_by(id: params[:user_id])
+    chat_room_user = ChatRoomUser.new(chat_room: @chat_room, user: user)
+
```

Comment:
微妙に修正が必要かもですが以下のようなイメージで書けませんかね？？
不要なselect文が出るので、できるだけ不要なselectが出ないようにしたいですね！
```suggestion
@chat_room = ChatRoom.find(params[:id])
chat_room_user = @chat_room.chat_room_users.new(user_id: params[:user_id])
```

Reply:
デバッグの結果は下のようになっていますが、このコードでオッケーでしょうか？
returnの書き方を+ returnのように書こうかと思いましたが、分からなかったのでこのようにしました。
```
    15: def add_user
    16:   binding.pry
 => 17:   @chat_room = ChatRoom.find(params[:id])
    18: 
    19:   return chat_room_user = @chat_room.chat_room_users.new(user_id: params[:user_id]) if params[:user_id].present?
    20: 
    21:   render json: { errors: chat_room_user.errors.full_messages }, status: 400 unless chat_room_user.save!
    22: end

[1] pry(#<ChatRoomsController>)> params[:id]
=> "3"
[2] pry(#<ChatRoomsController>)> ChatRoom.find(params[:id])
  ChatRoom Load (19.3ms)  SELECT `chat_rooms`.* FROM `chat_rooms` WHERE `chat_rooms`.`id` = 3 LIMIT 1
  ↳ (pry):6:in `add_user'
=> #<ChatRoom:0x0000ffff5e410650 id: 3, paid_or_free: true, created_at: Fri, 10 Nov 2023 08:48:18.600251000 UTC +00:00, updated_at: Fri, 10 Nov 2023 08:48:18.600251000 UTC +00:00>
[3] pry(#<ChatRoomsController>)> @chat_room = ChatRoom.find(params[:id])
  CACHE ChatRoom Load (0.1ms)  SELECT `chat_rooms`.* FROM `chat_rooms` WHERE `chat_rooms`.`id` = 3 LIMIT 1
  ↳ (pry):7:in `add_user'
=> #<ChatRoom:0x0000ffff5f336418 id: 3, paid_or_free: true, created_at: Fri, 10 Nov 2023 08:48:18.600251000 UTC +00:00, updated_at: Fri, 10 Nov 2023 08:48:18.600251000 UTC +00:00>
[4] pry(#<ChatRoomsController>)> params[:user_id].present?
=> true
[5] pry(#<ChatRoomsController>)> chat_room_user = @chat_room.chat_room_users.new(user_id: params[:user_id])
=> #<ChatRoomUser:0x0000ffff7e8360e8 id: nil, user_id: 21, chat_room_id: 3, created_at: nil, updated_at: nil>
[6] pry(#<ChatRoomsController>)> Started GET "/cable?uid=toshi.m3587@gmail.com&access-token=[FILTERED]&client=fVEgFtXb47MgCOcoZ2PJbA" for 192.168.65.1 at 2023-11-27 11:14:41 +0900
Started GET "/cable?uid=ma1@gmail.com&access-token=[FILTERED]&client=dHSiT35X9l5YakfoTzLDwA" for 192.168.65.1 at 2023-11-27 11:14:41 +0900
[6] pry(#<ChatRoomsController>)> chat_room_user.save!
  TRANSACTION (17.5ms)  BEGIN
  ↳ (pry):10:in `add_user'
  User Load (10.9ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 21 LIMIT 1
  ↳ (pry):10:in `add_user'
  ChatRoomUser Create (32.9ms)  INSERT INTO `chat_room_users` (`user_id`, `chat_room_id`, `created_at`, `updated_at`) VALUES (21, 3, '2023-11-27 02:20:12.267972', '2023-11-27 02:20:12.267972')
  ↳ (pry):10:in `add_user'
  TRANSACTION (10.8ms)  COMMIT
  ↳ (pry):10:in `add_user'
=> true
```

Reply:
これではダメですね

paramsにuser_idが存在していたら保存もせずかつrenderもせずreturnしてるので実質何もしてないコードになってます。suggestしたそのままで良いのではないかと思います。
また元々のコードもですが、保存が成功した時にrenderしてないので成功したら200を返すようにしましょう

指摘した内容としてはuser = User.find_by(id: params[:user_id])などのように毎回DBに問い合わせして、select文を使うと、DBに負荷がかかるのでアノテーションが使えるのであればアノテーションを使って、newすればDBへの問い合わせが減りますよという指摘になります！

```
@chat_room = ChatRoom.find(params[:id])
chat_room_user = @chat_room.chat_room_users.new(user_id: params[:user_id])
```